src/orange/OrangeParams.cc

```diff
@@ -125,22 +141,30 @@ OrangeParams::OrangeParams(OrangeInput input)
                       "stack is limited to a depth of "
                    << detail::LogicStack::max_stack_depth());
 
-    // TODO: update this to work over multiple universe levels
+    std::vector<Label> surface_labels;
+    std::vector<Label> volume_labels;
+
+    for (const UnitInput& u : input.units)
     {
         // Capture metadata
-        UnitInput& u = input.units.front();
-        surf_labels_ = LabelIdMultiMap<SurfaceId>{std::move(u.surfaces.labels)};
-        std::vector<Label> volume_labels;
-        volume_labels.resize(u.volumes.size());
-        for (auto i : range(u.volumes.size()))
-        {
-            volume_labels[i] = std::move(u.volumes[i].label);
-        }
-        vol_labels_ = LabelIdMultiMap<VolumeId>{std::move(volume_labels)};
-        bbox_       = u.bbox;
+
+        std::copy(u.surfaces.labels.begin(),
+                  u.surfaces.labels.end(),
+                  std::back_inserter(surface_labels));
+
+        std::transform(u.volumes.begin(),
+                       u.volumes.end(),
+                       std::back_inserter(volume_labels),
+                       [](VolumeInput v) -> Label { return v.label; });
```

Comment:
See the comment about volume labels: I thought we decided to uniquify on universe name if no "ext" is provided?